### DISCUSSION TITLE
Wazuh agent cant connect to the manager

### DISCUSSION THREAD

**Discussion by ghost** [Security-Onion-Solutions/securityonion#7658] [Unsupported Versions] [answered]
Hello, 

Just deployed a distributed version of SO, then installed an agent on a server, but it cant connect to the manager: 

`2022/03/28 16:03:52 ossec-agentd: INFO: Trying to connect to server (xxx.xxx.xxx.xxx:1514/udp).`

A netcat to 1515 works, but not the 1514 so it doesnt look like a fw issue. 

On the manager, a netcat to localhost 1514 is alright. 

Since, I'm new to SO, and Wazuh, I dont know what I'm missing exactly ...

**Comment by ghost** [comment]
Actually, on the manager, doing a netcat on its up doesnt work. It only works if I use localhost :thinking:

**Comment by dougburks** [comment, marked as answer]
Did you run `so-allow` to allow the traffic through the firewall?
https://docs.securityonion.net/en/2.3/wazuh.html#adding-agents
https://docs.securityonion.net/en/2.3/so-allow.html#so-allow

**Reply by ghost** [reply]
@dougburks Yeah did both. From the server where the agent is installed, I can netcat port 1515 but not 1514. 

On the manager node, nc -vz localhost 1514 works, but nc -vz xxx.xxx.xxx.xxx 1514, where that's the ip of that node, doesnt work. 
Is there a thing that avoid agents to connect to that ip ?

**Reply by dougburks** [reply]
One thing to note is that Wazuh port 1514 defaults to udp instead of tcp.

**Reply by ghost** [reply]
Yeah I know, I tried both but still got the same results :/

**Reply by dougburks** [reply]
Is Wazuh running?

Have you tried restarting Wazuh?

**Reply by ghost** [reply]
@dougburks Actually, found the issue in the wazuh container logs. 

In `/nsm/wazuh/etc/ossec.conf` in the remote part I had a <local_ip> that wasnt supposed to be there. 
Then, I was able to connect to the manager.

Last question, salt versions on all nodes should be the same or can we have different ones ?

**Reply by dougburks** [reply]
Salt versions on all nodes should be the same. If you have further questions or problems, please start a new discussion rather than replying to this Wazuh discussion. I'm marking this discussion as answered.

**Reply by ghost** [reply]
Thanks I will ! sorry for that :+1:
